### PR TITLE
Added headers parameters for ADS Langchain  

### PR DESCRIPTION
**Pull request by lu-ohai**
### Added headers parameters ADS Langchain

- The following classes will allow users to pass new parameters `headers`
-- ChatOCIModelDeployment
-- ChatOCIModelDeploymentVLLM
-- ChatOCIModelDeploymentTGI
-- BaseOCIModelDeployment
-- OCIModelDeploymentLLM
-- OCIModelDeploymentTGI
-- OCIModelDeploymentVLLM

Another PR will be made to `langchain_community` when this one is finalized 

### Notebook

Screenshot 2024-12-11 at 1 27 42 PM
Screenshot 2024-12-11 at 1 28 23 PM

### CODE REVIEW

**Review by qiuosier — DISMISSED** [review]
Your test examples shows that the header is pretty much useless :) It is showing an error whenever the default header is not used.

**Comment by lu-ohai** [comment]
> Your test examples shows that the header is pretty much useless :) It is showing an error whenever the default header is not used.

The proposed behavior is that users don't need to use (know) this parameter when they're calling the langchain class. This is added for future extension of feature. @mrDzurb @mayoor

File: ads/llm/langchain/plugins/llms/oci_data_science_model_deployment_endpoint.py [hunk 1]
```diff
@@ -125,12 +127,12 @@ def _headers(
         Returns:
             Dict: A dictionary containing the appropriate headers for the request.
         """
+        headers = self.headers
```
Comment (mrDzurb) [inline ads/llm/langchain/plugins/llms/oci_data_science_model_deployment_endpoint.py:130]: ```headers = self.headers or {}```?

File: ads/llm/langchain/plugins/chat_models/oci_data_science.py [hunk 2]
```diff
@@ -257,6 +263,9 @@ def _construct_json_body(self, messages: list, params: dict) -> dict:
     """Stop words to use when generating. Model output is cut off
     at the first occurrence of any of these substrings."""
 
+    headers: Optional[Dict[str, Any]] = {"route": DEFAULT_INFERENCE_ENDPOINT_CHAT}
```
[the remark this answers is not in the document]
Comment (lu-ohai) [reply ads/llm/langchain/plugins/chat_models/oci_data_science.py]: Just updated.

**Review by mrDzurb — APPROVED** [review]

**Review by qiuosier — APPROVED** [review]
For the user experience, `default_headers` seems to be a bit more complicated than simply using `headers`.

**Pull request merged**